### PR TITLE
Initial test suite

### PR DESCRIPTION
**Pull request by samholmes**

### CODE REVIEW

File: test/updateFiles.test.ts [hunk 1]
```diff
@@ -0,0 +1,244 @@
...
+  })
+
+  const isUpdateFilesResponse = (res: Response): void => {
+    isSuccessfulResponse(res)
+
+    expect(res.body.data, 'res.body.data').to.be.an('object')
+    expect(res.body.data.timestamp, 'res.body.data.timestamp').to.be.a('number')
+    expect(res.body.data.paths, 'res.body.data.paths').to.be.an('object')
+
+    repoTimestamp = res.body.data.timestamp
```
Comment (samholmes) [inline test/updateFiles.test.ts:31]: Move this to it's own function for clarity (`updateRepoTimestamp`)

File: test/updateFiles.test.ts [hunk 2]
```diff
@@ -0,0 +1,244 @@
...
+          [filePath]: { text: 'content' }
+        }
+      })
+      .expect(
+        isErrorResponse(
+          422,
+          `Unable to write files under '${directoryPath}'. ` +
+            `Existing document is not a directory.`
+        )
+      )
```
Comment (samholmes) [inline test/updateFiles.test.ts:176]: This test should not depend on `/file` existing. Better to create a new file and then try to write a file under that new file all within the test.

File: test/updateFiles.test.ts [hunk 3]
```diff
@@ -0,0 +1,244 @@
...
+    await agent
+      .post('/api/v3/updateFiles')
+      .send({
+        repoId,
+        timestamp: repoTimestamp,
+        paths: {
+          '/file': null
+        }
+      })
+      .expect(isUpdateFilesResponse)
```
Comment (samholmes) [inline test/updateFiles.test.ts:189]: This should also create the file before deleting it. (No interdependencies between tests)

File: test/updateFiles.test.ts [hunk 4]
```diff
@@ -0,0 +1,244 @@
...
+        paths: {
+          [filePath]: null
+        }
+      })
+      .expect(
+        isErrorResponse(
+          422,
+          `Unable to delete file '${filePath}'. ` + `File is already deleted.`
+        )
+      )
```
Comment (samholmes) [inline test/updateFiles.test.ts:229]: No interdependencies again

**Review by thehobbit85 — APPROVED** [review]

**Pull request merged**